# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ModuleMapKey.h
# Commit: 85fd9fc9cffa
# Full Hash: 85fd9fc9cffa98039328796d30a2d10386283b3e
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-21 21:39:46
# Regressor Bug: 1742437
# File Overlap Count: 2
# Description:
#   Bug 1742437 - Move module loader files into a central folder; r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D137532
# ==============================================================================

diff -r 7ecc4f9b20ad -r 85fd9fc9cffa dom/script/ModuleMapKey.h
--- a/dom/script/ModuleMapKey.h	Mon Feb 21 15:39:08 2022 +0000
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,47 +0,0 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-#ifndef ModuleMapKey_h__
-#define ModuleMapKey_h__
-
-#include "nsURIHashKey.h"
-#include "nsIGlobalObject.h"
-
-namespace mozilla::dom {
-
-class ModuleMapKey : public nsURIHashKey {
- public:
-  using KeyType = ModuleMapKey&;
-  using KeyTypePointer = const ModuleMapKey*;
-
-  ModuleMapKey() = default;
-  ModuleMapKey(const nsIURI* aURI, nsIGlobalObject* aWebExtGlobal);
-  explicit ModuleMapKey(const ModuleMapKey* aKey);
-  ModuleMapKey(ModuleMapKey&& aToMove) noexcept;
-
-  ModuleMapKey& operator=(const ModuleMapKey& aOther) = default;
-
-  KeyType GetKey() { return *this; }
-  KeyTypePointer GetKeyPointer() const { return this; }
-  static KeyTypePointer KeyToPointer(KeyType aKey) { return &aKey; }
-
-  bool KeyEquals(KeyTypePointer aOther) const;
-  static PLDHashNumber HashKey(KeyTypePointer aKey);
-
-  nsCOMPtr<nsIGlobalObject> mWebExtGlobal;
-};
-
-inline void ImplCycleCollectionUnlink(ModuleMapKey& aField) {
-  ImplCycleCollectionUnlink(aField.mWebExtGlobal);
-}
-
-inline void ImplCycleCollectionTraverse(
-    nsCycleCollectionTraversalCallback& aCallback, ModuleMapKey& aField,
-    const char* aName, uint32_t aFlags = 0) {
-  ImplCycleCollectionTraverse(aCallback, aField.mWebExtGlobal, aName, aFlags);
-}
-
-}  // namespace mozilla::dom
-
-#endif